src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,225 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+        } else {
+            last.next = new Node<>(last, value, null);
+            last = last.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        indexAddValidatorForAddElement(index);
+
+        if (index == 0 && size == 0) {
+            addElementToEmptyList(value, index);
+            return;
+        }
+
+        if (index == 0) {
+            addElementToStart(value);
+            return;
+        }
+
+        if (index == size) {
+            add(value);
+            return;
+        }
+        addElementToMiddle(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (T element : list) {
+                add(element);
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        Node<T> currentNode = first;
+        if (index == 0) {
+            return currentNode.item;
+        }
+
+        for (int i = 0; i < index; i++) {
+            currentNode = currentNode.next;
+        }
```

Comment:
[Common mistakes](https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/linked-list.html#dont-iterate-your-nodes-from-the-head-to-tail-or-vice-versa)
**Don’t iterate your nodes from the head to tail or vice versa.**